### DISCUSSION TITLE
Suggestion 快捷指令 和 回车发送冲突

### DISCUSSION THREAD

**Discussion by mashupeng1** [ant-design/x#359] [Q&A]
![image](https://github.com/user-attachments/assets/d87ba6bc-7ae1-403d-bfaf-4f1d7c294cd2)
当我/ 弹出 suggestion list后，我键盘选择，回车点选一个满意的suggestion item后，我想选择，但是同时也把我的/发送出去了。这怎么避免呢？
![image](https://github.com/user-attachments/assets/12e729af-0e75-4239-afc5-e7c2828fe885)

**Comment by petercat-assistant** [comment]
在使用 suggestion 列表时，您遇到了快捷指令和回车发送冲突的问题。通常，这种情况可能是因为在选择 suggestion 时，回车键的默认行为是触发提交操作。为了避免这种情况，您可以尝试以下方法：

1. **自定义回车键行为**：在处理 suggestion 选择时，拦截回车键的默认行为，只执行选择操作，而不触发提交。

2. **使用其他键进行选择**：如果可能，您可以配置使用其他键（例如 Tab 键）来选择 suggestion，而保留回车键用于提交。

3. **检查组件配置**：查看 ant-design/x 组件的配置选项，看看是否有相关的设置可以调整回车键的行为。

如果这些方法无法解决问题，可能需要查看具体的代码实现或配置，或者在 ant-design/x 的 GitHub 讨论区中查看是否有类似的问题和解决方案。

如果您需要进一步的帮助，请在评论中描述您的问题并 @petercat-assistant 以便与我开始对话。

**Comment by YumoImer** [comment]
`Sender` 组件的 `submitType` 默认为 `'enter'`, 可以配置为 `'shiftEnter'` 避免在按回车的时候触发 `onSubmit`

**Comment by ztkuaikuai** [comment]
也许你可以通过以下步骤来实现你的需求，而不必将 `submitType` 配置为 `'shiftEnter'`，因为我觉得从用户视角看使用 `'shiftEnter'` 输出非常反直觉：
1. 通过 `Suggestion` 组件中的 `onOpenChange` 方法获取到面板是否打开的状态，在自己的组件中维护该状态；
2. 在 `Sender` 组件中 `onSubmit` 事件里做前置判断，如果面板打开，就不做提交。

以下是一个简单的示例
```js
const handleSubmit = (value: string) => {
    if (isSuggestionOpen) return
    setMessages(prev => [...prev, { content: value, role: 'user' }])
    setValue('')
}
```
